# *Brassica rapa*[芜菁]
+ 基因组大小 ： 485 Mb - [《The genome of the mesopolyploid crop species Brassica rapa - nature genetics》](https://www.nature.com/articles/ng.919)
+ 叶绿体基因组[CN_015139.1](https://www.ncbi.nlm.nih.gov/nuccore/323481965) - 153482 bp
+ 线粒体基因组[JF920285](https://www.ncbi.nlm.nih.gov/nuccore/JF920285.1) - 219747 bp


## *Brassica rapa* type
| 拉丁名 | 俗称 | 中文名 | 图片 |
| --- | --- | --- | --- |
| B. rapa subsp. rapa | turnip | 芜菁 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/rapa.png) | 
| B. rapa ssp. chinensis | nonheading Chinese cabbage | 白菜/青菜/小白菜 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/chinensis.png) | 
| B. rapa ssp. pekinenesis | heading Chinese cabbage | 大白菜 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/pekinenesis.png) |
| B. rapa var. glabra | | 白菜 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/glabra.png) |
| B. rapa subsp. campestris | winter rape | 芸薹 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/campestris.png) |
| B. rapa var parachinensis | choy sum | 菜心 |
| B. rapa var purpuraria | purple pakchoi | 紫菜薹 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/purpuraria.png) |
| B. rapa ssp. oleifera | baby bokchoi | 油白菜 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/oleifera.png) |
| B. rapa ssp. narinosa | broadbeaked mustard | 塌棵菜 | ![](https://github.com/eternal-bug/Sequencing-and-assembly/blob/master/plastid/Brassicaceae/Brassica/pic/narinosa.png) |  
| B. rapa var perviridis | komatsuna | 小松菜（コマツナ） |
| B. rapa ssp. nipposinica | mizuna | 水菜 (ミズナ) |
| B. rapa ssp. rapifera |  | 芜菁 |
| B. rapa ssp. trilocularis | yellow sarson | 

![](https://media.nature.com/m685/nature-assets/hortres/2014/hortres201424/images_hires/hortres201424-f1.jpg)
(a) Morphotypes of B. rapa; top two lines from left to right: pak choi, heading B. rapa, turnip, oilseed, purple pak choi, caixin, mizuna, purple caitai and takucai; the third line shows additional morphotypes or varieties of the previous morphotypes. 

参考
+ [芸薹属，混乱的白菜家族，之一（“薹”字读音见内文）](http://www.vccoo.com/v/410ea4) 
+ [白菜、芥菜、卷心菜：祖上都是“外国人”](http://blog.sina.com.cn/s/blog_487ab9d50102v624.html)
+ [《Genome triplication drove the diversification of Brassica plants》](https://www.nature.com/articles/hortres201424)
+ [《biotechnology in agriculture and forestry》](https://books.google.com/books?id=66jvCAAAQBAJ&dq=Genome+analysis+in+Brassica+with+special+reference+to+the+experimental+formation+of+B.+napus+and+peculiar+mode+of+fertilization.&hl=zh-CN)
+ [PPBC - 芸薹属](http://www.plantphoto.cn/gen/257242)

## 相关
+ [Brassica rapa FPsc](https://fpsc.wisc.edu/growguide/Instructions_and_Tips_for_Growing_FPsc.pdf)
+ [fastplants](https://fastplants.org)

## 可能可用的数据
+ [PRJNA312457](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA312457)

## 项目1
+ 多样性较低
+ Arabidopsis comparative genomics(ACG) - Brassica, Boechera, Arabidopsis 
```
SRX2330877
SRX2330820
SRX2095535
SRX2095530
SRX2095529
SRX2095528
SRX2066845
SRX2066841
SRX2066838
SRX2066836
SRX2066834
SRX2066833
SRX2066832
SRX2066831
SRX2066830
SRX2066829
SRX2066828
SRX2066827
SRX2066826
SRX2066825
```

### SRR文件信息
| type | SRR | size.bp | coverage |
| --- | --- | --- | ---- |
| FPsc S7 B8 | SRR4996371 | 1,412,600,400 * 2 | 6 |
| FPsc S7 D4 | SRR4996316 | 1,108,828,950 * 2 | 5 |
| FPsc S7 B10 | SRR4116114 | 1,168,897,950 * 2 | 5 |
| FPsc S7 E9 | SRR4116071 | 813,978,150 * 2 | 3 |
| FPsc S7 D9 | SRR4116070 | 1,290,799,950 * 2 | 5 |
| FPsc S7 G10 | SRR4116069 | 1,135,376,100 * 2 | 5 |
| FPsc S7 H10 | SRR4098140 | 1,334,516,400 * 2 | 5 |
| FPsc S7 H11 | SRR4098136 | 1,170,578,550 * 2 | 5 |
| FPsc S7 G11 | SRR4098135 | 1,317,606,450 * 2 | 5 |
| FPsc S7 G10 | SRR4098134 | 1,396,111,200 * 2 | 5 |
| FPsc S7 G7  | SRR4098132 | 3,972,200,413 * 2 | 16|
| FPsc S7 E11 | SRR4098131 | 1,163,692,500 * 2 | 5 |
| FPsc S7 G1  | SRR4098130 | 3,649,365,584 * 2 | 15 |
| FPsc S7 F9  | SRR4098129 | 1,217,231,550 * 2 | 5 |
| FPsc S7 G9  | SRR4098128 | 641,326,800 * 2 | 2 |
| FPsc S7 D10 | SRR4098127 | 1,582,516,200 * 2 | 6 |
| FPsc S7 F11 | SRR4098126 | 1,816,539,300 * 2 | 7 |
| FPsc S7 E10 | SRR4098125 | 1,236,223,650 * 2 | 5 |
| FPsc S7 H9  | SRR4098124 | 973,244,400 * 2 | 4 |
| FPsc S7 G12 | SRR4098123 | 1,112,849,700 * 2 | 4 |

## 项目2
+ PRJNA421038
+ [《Identification of Functional Single-Nucleotide Polymorphisms Affecting Leaf Hair Number in Brassica rapa》](http://www.plantphysiol.org/content/177/2/490)
+ 包含较多亚种 B. rapa ssp. chinensis, B. rapa var parachinensis, B. rapa var purpuraria, B. rapa ssp. oleifera, B. rapa ssp. narinosa, B. rapa var perviridis, and B. rapa ssp. nipposinica. 
+ [Run信息](https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA421038&go=go)

### 项目2的Run
| type | SRR | size.bp | coverage |
| --- | --- | --- | ---- |
| SRR7006174 | 4,355,496,300 * 2 | 72 |
| SRR7006175 | 2,483,232,600 * 2 | 41 |
| SRR7006176 | 3,524,927,850 * 2 | 58 |
| SRR7006177 | 2,760,999,450 * 2 | 46 |
| SRR7006178 | 3,059,902,800 * 2 | 50 |
| SRR7006179 | 2,686,803,150 * 2 | 44 |
| SRR7006180 | 2,648,847,000 * 2 | 44 |
| SRR7006181 | 3,492,660,000 * 2 | 58 |
| SRR7006182 | 2,791,469,400 * 2 | 46 |
| SRR7006183 | 2,685,778,950 * 2 | 44 |
| SRR7006184 | 2,980,282,650 * 2 | 49 |
| SRR7006186 | 2,903,622,750 * 2 | 48 |
| SRR7006187 | 2,458,528,500 * 2 | 41 |
| SRR7006188 | 2,484,051,450 * 2 | 41 |
| SRR7006189 | 2,571,679,950 * 2 | 42 |


## 运行

### SRR4996371
+ 6

```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4996371
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 24 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4996316
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4996316
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4116114
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4116114
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4116071
+ 3
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4116071
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 12 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4116070
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4116070
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4116069
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4116069
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098140
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098140
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098136
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098136
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098135
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098135
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098134
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098134
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098132
+ 16
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098132
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 64 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098131
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098131
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098130
+ 15
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098130
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 60 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098129
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098129
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098128
+ 2
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098128
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 8 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098127
+ 6
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098127
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 24 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098126
+ 7
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098126
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 28 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098125
+ 5
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098125
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 20 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098124
+ 4
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098124
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 16 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### SRR4098123
+ 4
```bash
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
BASE_NAME=SRR4098123
cd ${WORKING_DIR}/${BASE_NAME}
bash 0_realClean.sh

anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff 16 --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

# 提交超算任务
bsub -q mpi -n 24 -J "${BASE_NAME}" "
  bash 0_master.sh
"
```

### qita
```bash
function echo_fastq_size {
  SRR=$1
  size=$2
  fold=$3
  format_size=$(echo ${size} | perl -MNumber::Format -n -e 'chomp;print Number::Format::format_number($_)')
  echo -e "| ${SRR} | ${format_size} * 2 | ${fold} |"
}

echo -n >srr_size_cov.txt
export genome_size=485000000
genome_file=genome.fa
WORKING_DIR=~/stq/data/anchr/Brassica_rapa
cd ${WORKING_DIR}
~/stq/Applications/biosoft/bwa-0.7.13/bwa index ./genome/genome.fa

list=($(ls -d SRR*))
((list_len=${}))

# get cut off fold number
n=0
fold_list=()
parallel -j 20 --ungroup -k --delay 1 "
  bash ~/stq/Applications/my/stat/stat_fastq_size.sh ./sequence_data/{1}_1.fastq.gz | tail -n 1 
" ::: ${list[@]} | \
sed 's/ //g' | \
sed 's/^|//g' | \
sed 's/|$//g' | \
sed 's/|/:/g' | \
while IFS=$':' read -r -a row;
do
  SRR=$(basename ${row[0]} | perl -p -e 's/[._]R?\d+\.f(ast)*q\.gz//')
  NUMBER=${row[1]}
  DEPTH=$(echo ${NUMBER} | sed 's/,//g' | perl -n -e 'printf "%.0f",$_*2/$ENV{genome_size}')
  FOLD=$(echo ${NUMBER} | sed 's/,//g' | perl -n -e 'printf "%.0f",$_*2*4/$ENV{genome_size}')
  echo ${SRR}\|${NUMBER}\|${DEPTH}\|${FOLD}
done | \
sort -k1.4 -t\| > srr_size_cov.txt
# 管道的while是在子shell中的，无法修改父进程的列表。所以不能从管道中读取数据
while read i
do
  num=$( echo ${i} | cut -f4 -d\| | sed 's/,//g' )
  fold_list[${n}]=${num}
  ((n++))
done < srr_size_cov.txt

((max=${#list[@]}-1))
for i in $(seq 0 ${max});
do
  BASE_NAME=${list[$i]}
  fold=${fold_list[$i]}
  cd ${WORKING_DIR}/${BASE_NAME}
  anchr template \
    . \
    --basename ${BASE_NAME} \
    --queue mpi \
    --genome 1_000_000 \
    --fastqc \
    --kmergenie \
    --insertsize \
    --sgapreqc \
    --trim2 "--dedupe --cutoff ${fold} --cutk 31" \
    --qual2 "25" \
    --len2 "60" \
    --filter "adapter,phix,artifact" \
    --mergereads \
    --ecphase "1,2,3" \
    --cov2 "40 80 120 160 240 320" \
    --tadpole \
    --splitp 100 \
    --statp 1 \
    --fillanchor \
    --xmx 110g \
    --parallel 24

  bsub -q mpi -n 24 -J "${BASE_NAME}" '
     bash 0_cleanup.sh
     bash 0_master.sh
     ~/stq/Applications/biosoft/bwa-0.7.13/bwa mem \
         -t 20 \
         -M   \
         ../genome/genome.fa \
         ./2_illumina/trim/Q25L60/R1.fq.gz \
         ./2_illumina/trim/Q25L60/R2.fq.gz > ./align/Rp.sam
     ~/stq/Applications/biosoft/bwa-0.7.13/bwa mem \
         -t 20 \
         -M   \
         ../genome/genome.fa \
         ./2_illumina/trim/Q25L60/Rs.fq.gz > ./align/Rs.sam
         
     cp ./align/Rp.sam ./align/R.sam
     cat ./align/Rs.sam | grep -v "^@" >> ./align/R.sam
     samtools view -b -o ./align/R.bam ./align/R.sam
     samtools sort -o ./align/R.sort.bam ./align/R.bam
     samtools index ./align/R.sort.bam
     rm ./align/*.sam
  '
done
```
